src/server/editorServices.ts

```diff
@@ -1463,11 +1734,23 @@ namespace ts.server {
             let configFileName: NormalizedPath;
             let configFileErrors: Diagnostic[];
 
+            const info = this.getOrCreateScriptInfoForNormalizedPath(fileName, /*openedByClient*/ true, fileContent, scriptKind, hasMixedContent);
             let project: ConfiguredProject | ExternalProject = this.findContainingExternalProject(fileName);
             if (!project) {
-                ({ configFileName, configFileErrors } = this.openOrUpdateConfiguredProjectForFile(fileName, projectRootPath));
+                configFileName = this.getConfigFileNameForFile(info, projectRootPath);
                 if (configFileName) {
                     project = this.findConfiguredProjectByProjectName(configFileName);
+                    if (!project) {
+                        project = this.openConfigFile(configFileName, fileName);
+
+                        // even if opening config file was successful, it could still
+                        // contain errors that were tolerated.
+                        const errors = project.getGlobalProjectErrors();
+                        if (errors && errors.length > 0) {
+                            // set configFileErrors only when the errors array is non-empty
+                            configFileErrors = errors;

```

Comment:
It looks like we won't set `configFileErrors` if the project already existed, but couldn't it still have errors?